modules/appierBidAdapter.js

```diff
@@ -0,0 +1,89 @@
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER } from 'src/mediaTypes';
+import { config } from 'src/config';
+
+export const ADAPTER_VERSION = '1.0.0';
+const SUPPORTED_AD_TYPES = [BANNER];
+
+// we have different servers for different regions / farms
+export const API_SERVERS_MAP = {
+  'default': 'ad2.apx.appier.net',
+  'tw': 'ad2.apx.appier.net',
+  'jp': 'ad-jp.apx.appier.net'
+};
+
+const BIDDER_API_ENDPOINT = '/v1/prebid/bid';
+
+export const spec = {
+  code: 'appier',
+  supportedMediaTypes: SUPPORTED_AD_TYPES,
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    return typeof bid.params.hzid === 'string';
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {bidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (bidRequests, bidderRequest) {
+    if (bidRequests.length === 0) {
+      return [];
+    }
+    const server = this.getApiServer();
+    const bidderApiUrl = `//${server}${BIDDER_API_ENDPOINT}`
+    const payload = {
+      'bids': bidRequests,
+      'refererInfo': bidderRequest.refererInfo,
+      'version': ADAPTER_VERSION
+    };
+    return [{
+      method: 'POST',
+      url: bidderApiUrl,
+      data: payload,
+      // keep the bidder request object for later use
+      bidderRequest: bidderRequest
+    }];
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {serverResponse} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, serverRequest) {
+    if (!serverResponse.body || !Array.isArray(serverResponse.body)) {
```

Comment:
You can optimize this line to:
```
if (!Array.isArray(serverResponse.body)) {
  return []
}
```

Reply:
@idettman I did the requested change and did git squash.
Thanks for the prompt review and the suggestions!